app/uk/gov/hmrc/tai/model/domain/TaxComponentType.scala

```diff
@@ -186,50 +186,52 @@ case object JobSeekerAllowanceIncome extends TaxCodeIncomeComponentType
-    "UkDividend" -> UkDividend,
-    "UnitTrust" -> UnitTrust,
-    "StockDividend" -> StockDividend,
-    "NationalSavings" -> NationalSavings,
-    "SavingsBond" -> SavingsBond,
-    "PurchasedLifeAnnuities" -> PurchasedLifeAnnuities,
-    "UntaxedInterestIncome" -> UntaxedInterestIncome,
-    "IncapacityBenefit" -> IncapacityBenefit,
-    "JobSeekersAllowance" -> JobSeekersAllowance,
-    "EmploymentAndSupportAllowance" -> EmploymentAndSupportAllowance
-  )
-
-  implicit val formatTaxComponentType = new Format[NonTaxCodeIncomeComponentType] {
-    override def reads(json: JsValue): JsSuccess[NonTaxCodeIncomeComponentType] = JsSuccess(nonTaxCodeIncomesMap.getOrElse(json.as[String],
-      throw new IllegalArgumentException("Invalid Non tax code component type")))
-
+  implicit val formatTaxComponentType: Format[NonTaxCodeIncomeComponentType] = new Format[NonTaxCodeIncomeComponentType] {
+    override def reads(json: JsValue): JsSuccess[NonTaxCodeIncomeComponentType] = json.as[String] match {
+      case "NonCodedIncome" => JsSuccess(NonCodedIncome)
+      case "Commission" => JsSuccess(Commission)
+      case "OtherIncomeEarned" => JsSuccess(OtherIncomeEarned)
+      case "OtherIncomeNotEarned" => JsSuccess(OtherIncomeNotEarned)
+      case "PartTimeEarnings" => JsSuccess(PartTimeEarnings)
+      case "Tips" => JsSuccess(Tips)
+      case "OtherEarnings" => JsSuccess(OtherEarnings)
+      case "CasualEarnings" => JsSuccess(CasualEarnings)
+      case "ForeignDividendIncome" => JsSuccess(ForeignDividendIncome)
+      case "ForeignPropertyIncome" => JsSuccess(ForeignPropertyIncome)
+      case "ForeignInterestAndOtherSavings" => JsSuccess(ForeignInterestAndOtherSavings)
+      case "ForeignPensionsAndOtherIncome" => JsSuccess(ForeignPensionsAndOtherIncome)
+      case "StatePension" => JsSuccess(StatePension)
+      case "OccupationalPension" => JsSuccess(OccupationalPension)
+      case "PublicServicesPension" => JsSuccess(PublicServicesPension)
+      case "ForcesPension" => JsSuccess(ForcesPension)
+      case "PersonalPensionAnnuity" => JsSuccess(PersonalPensionAnnuity)
+      case "Profit" => JsSuccess(Profit)
+      case "BankOrBuildingSocietyInterest" => JsSuccess(BankOrBuildingSocietyInterest)
+      case "UkDividend" => JsSuccess(UkDividend)
+      case "UnitTrust" => JsSuccess(UnitTrust)
+      case "StockDividend" => JsSuccess(StockDividend)
+      case "NationalSavings" => JsSuccess(NationalSavings)
+      case "SavingsBond" => JsSuccess(SavingsBond)
+      case "PurchasedLifeAnnuities" => JsSuccess(PurchasedLifeAnnuities)
+      case "UntaxedInterestIncome" => JsSuccess(UntaxedInterestIncome)
+      case "IncapacityBenefit" => JsSuccess(IncapacityBenefit)
+      case "JobSeekersAllowance" => JsSuccess(JobSeekersAllowance)
+      case "EmploymentAndSupportAllowance" => JsSuccess(EmploymentAndSupportAllowance)
+      case _ => throw new IllegalArgumentException("Invalid Non tax code component type")
+    }
```

Comment:
This is missing a test, I'd be okay with just testing one happy case and the exception case

Reply:
NonTaxCodeIncomeComponentType is not missing tests

I have added some tests for TaxCodeIncomeComponentType though